### ISSUE TITLE
Failing one portion of a batch craft makes the rest of the batch useless (stuck in progress)

### ISSUE THREAD

**Issue by PlatoSoft** [CleverRaven/Cataclysm-DDA#42851]
# Describe the bug
I made a batch of baked beans using 20 bratwurst. At 85% one of the bratwursts were destroyed by a failed craft.
Now the entire batch is useless, because I don't have any more bratwurst. All of the baked beans I made up until 85% is locked inside the "in progress (85%)" item.

# Steps To Reproduce
1. Try to batch craft something and hope for failure (tedious)

# Expected behavior
Being able to use the beans that were completed.

# Screenshots
No screenshots needed, i suppose.

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-4951-g9af1a35 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    Blaze Industries [blazeindustries],
    No Rail Stations [No_Rail_Stations],
    Folding Parts pack [deoxymod],
    Graphical Overmap [Graphical_Overmap],
    Disable NPC Needs [no_npc_food],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context
I suppose this COULD count as a suggestion, but I think that it counts as a bug since I did complete most of the craft. 
Discuss if you disagree.
This is my first issue on here, let me me know if I made a mistake.
Cheers

**Comment by wapcaplet** [comment]
I agree, this behavior has bothered me for as long as I can remember. It affects all crafting recipes, even when not using batch mode - if the crafting is interrupted for any reason, you are left with an in-progress item that cannot be used until it's finished. For tools and such, this makes sense, but for food it's a little strange. A half-finished longbow is rightfully worthless, but half-baked beans or half-brewed tea still ought to be edible, if lacking in nutritional value or taste.

Given the way the crafting progress item works, this could be technically challenging to solve in a general sense for all crafting, but I wonder if there's an opportunity here to improve the batch-mode crafting to allow success with a smaller batch size.

For example, when you try to batch-craft 20 items, and you fail and ruin some of the components, it would be nice to have a fallback option, say "You no longer have enough ingredients to make a batch of 20, but you can still finish crafting 18 of them. Proceed? (Y/N)" Answering "Y" would attempt to finish the 18 (and possibly fail and fall back again), while "N" would simply quit and leave you with the in-progress batch of 20, as it does now.

**Comment by PlatoSoft** [comment]
Yes, an option to continue with the remaining batch would be great.

**Comment by Proluiz** [comment]
my opinion there should be a 3th option on it, because when you mess up and lose some of the components of the craft the game still ask you if you want to use any extra of your recourcers to finish it, this should be kept as an option too.
